### PR TITLE
METRON-633: Create better logging for HbaseEnrichmentWriter

### PR DESCRIPTION
**Pull request by zezutom**
## Contributor Comments
I have added logging to critical paths, so that issues can be tracked down more easily.

## Pull Request Checklist
In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel). 
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

### For code changes:
- [x] Have you ensured that the full suite of tests and checks have been executed in the root incubating-metron folder via:
  ```
  mvn -q clean integration-test install && build_utils/verify_licenses.sh 
  ```

- [x] Have you written or updated unit tests and or integration tests to verify your changes?

### CODE REVIEW

**Comment by zezutom** [comment]
Hey folks, this one has been open for a while. Could I please get some +1's? Thank you!

**Comment by justinleet** [comment]
Can you use parameterized messages for the various log statements or `logger.isDebugEnabled()`, rather than constructing the strings every time?  The common case is for this to not be run, so I'd rather not have this stuff constructed to much. See: https://www.slf4j.org/faq.html#logging_performance

If I recall correctly, typically it's better to use parameterized messages if it's just using the `toString()` of an object, and `logger.isDebugEnabled()` if not (to avoid whatever calls are made).  I'd have to double check that though.

**Comment by mmiklavc** [comment]
Agreed with @justinleet about switching away from isDebugEnabled(). Except for instances where you're doing something that has a fair amount of overhead in evaluating the pre-formatted strings, using message parameters is the current recommended approach. Also allow me to plug my PR that normalizes logging across the board - https://github.com/apache/metron/pull/599 :-)

File: metron-platform/metron-enrichment/src/main/java/org/apache/metron/enrichment/writer/SimpleHbaseEnrichmentWriter.java [hunk 1]
```diff
@@ -169,16 +188,19 @@ public HTableInterface getTable(Map<String, Object> config) throws IOException {
   private List<String> getColumns(Object keyColumnsObj, boolean allowNull) {
     Object o = keyColumnsObj;
     if(allowNull && keyColumnsObj == null) {
+      LOG.debug("No key columns were specified");
       return Collections.emptyList();
     }
     if(o instanceof String) {
+      LOG.debug("Key column: '" + o + "'");
```
Comment (mmiklavc) [inline metron-platform/metron-enrichment/src/main/java/org/apache/metron/enrichment/writer/SimpleHbaseEnrichmentWriter.java]: Better to turn this and similar statements into `LOG.debug("Key column: '{}'", o );`. Per the slf4j docs, this has the potential to improve calls by up to 30x where debugging is not enabled.

Comment (zezutom) [inline metron-platform/metron-enrichment/src/main/java/org/apache/metron/enrichment/writer/SimpleHbaseEnrichmentWriter.java]: Thanks everyone for a great feedback! @justinleet I will rebase once #599 is merged, cheers.

**Comment by cestella** [comment]
Just piling in here and saying thanks @zezutom for the contribution.  This looks great!

**Comment by zezutom** [comment]
Thank you @cestella and sorry for the delay, now I am back to it.

**Comment by justinleet** [comment]
@zezutom When you have a chance, could you merge in master and deconflict this?  The Travis build timeouts should be fixed as of https://github.com/apache/metron/pull/624.  Sorry about the trouble with Travis, and I look forward to getting this into master.

**Comment by zezutom** [comment]
@justinleet Will do, thanks for sorting out Travis!

**Comment by zezutom** [comment]
@justinleet I can't merge this PR, do you mind merging to master yourself? Cheers.

**Comment by justinleet** [comment]
+1, thanks again for the contribution.

**Pull request closed without merging**